src/sleep/sleep.c

```diff
@@ -166,9 +203,10 @@ static int execute(char **modes, char **states) {
 
         /* Configure the hibernation mode */
         if (!strv_isempty(modes)) {
-                r = write_hibernate_location_info();
+                r = configure_hibernation();
                 if (r < 0)
-                        return log_error_errno(r, "Failed to write hibernation disk offset: %m");
+                        return log_error_errno(errno, "Failed to prepare for hibernation: %m");
```

Comment:
our functions are supposed to return negative errno-values as return value, and not via the actual `errno` variable. This change which replaces `r` by `errno` is hence not right.

See https://systemd.io/CODING_STYLE#error-handling for details.